apps/base-docs/base-camp/docs/hardhat-contract-to-contract/hardhat-contract-to-contract.md

```diff
@@ -0,0 +1,260 @@
+---
+title: 'Contract to contract interaction'
+description: Interact with other smart contracts
+hide_table_of_contents: false
+---
+
+In this article, you'll learn how to interact with other smart contracts using interfaces and the `.call()` function, which allows you to interact with other smart contracts without using an interface.
+
+---
+
+## Objectives
+
+By the end of this lesson you should be able to:
+
+- Use interfaces to allow a smart contract to call functions in another smart contract
+- Use the `call()` function to interact with another contract without using an interface
+
+---
+
+## Overview
+
+Interacting with external smart contracts is a very common task in the life of a smart contract developer. This includes interacting with contracts that are already deployed to a particular network.
+
+Usually the creators of certain smart contracts document their functionality and expose their functions by providing interfaces that can be used to integrate those particular contracts into your own.
+
+For instance, [Uniswap] provides documentation on how to interact with their smart contracts and also some packages to easily integrate their protocol.
+
+In this example, you interact with the [Uniswap protocol] to create a custom pool for a custom pair of tokens.
+
+Since the Uniswap protocol is already deployed, you will use [Hardhat forking] to test your contract.
+
+You will also use the following two approaches in the example:
+
+- Using interfaces
+- Using the `.call()` function
+
+## Interacting with deployed contracts using interfaces
+
+You must first install the [Uniswap V3 core package] by running:
+
+```bash
+npm install @uniswap/v3-core
+```
+
+This package provides access to the Uniswap interfaces of the Core protocol.
+
+Then, write a custom contract called `PoolCreator` with the following code:
+
+```typescript
```

Comment:
@latechwriter anything with:  // SPDX-License-Identifier: UNLICENSED and/or pragma solidity ^0.8.19; should have the language type switched to `solidity` please and thanks.  

When in doubt, please leave it as typescript as parts of this will have it too.